pkg/vpc/vpc.go

```diff
@@ -276,3 +368,38 @@ func cleanupSubnets(spec *api.ClusterConfig) {
 		}
 	}
 }
+
+func validatePublicSubnet(subnets []*ec2.Subnet) error {
+	for _, sn := range subnets {
+		if sn.MapPublicIpOnLaunch == nil || !*sn.MapPublicIpOnLaunch {
+			return fmt.Errorf("found mis-configured subnet %q. Expected public subnet with property "+
+				"\"MapPublicIpOnLaunch\" enabled. Without it new nodes won't get an IP assigned", *sn.SubnetId)
+		}
+	}
+	return nil
+}
+
+func SelectPublicNodeGroupSubnets(nodegroupAZs []string, clusterSpec *api.ClusterConfig) ([]string, error) {
+	numNodeGroupsAZs := len(nodegroupAZs)
+	if numNodeGroupsAZs == 0 {
+		return make([]string, 0), nil
```

Comment:
Minor nit: can also be just
```suggestion
		return nil, nil
```

and the calling code will still work.

Reply:
thanks!